### PR TITLE
Support Django 4.1, Python 3.10, and fix Selenium tests

### PR DESCRIPTION
**Pull request by adamchainz**
1. Add Django 4.1 to the test matrix
2. Add Python 3.10 to the test matrix
3. Deal with some deprecation warnings from running the tests on Django 4.0+
4. Update Selenium method calls for [removals in Selenium 4.3.0](https://github.com/SeleniumHQ/selenium/blob/trunk/py/CHANGES) of `find_element[s]_by_*` methods, to the new `find_element[s]` method with `By` argument ([docs](https://selenium-python.readthedocs.io/locating-elements.html)).

### CODE REVIEW

File: tests/testapp/settings.py [hunk 1]
```diff
@@ -37,5 +39,7 @@
 
 SECRET_KEY = "123456"
 
-USE_L10N = True
+if django.VERSION < (4, 0):
+    USE_L10N = True
```
Comment (adamchainz) [inline tests/testapp/settings.py:43]: fix deprecation warning about [setting removal](https://docs.djangoproject.com/en/4.0/releases/4.0/#localization):

```
/.../django-select2/venv/lib/python3.10/site-packages/django/conf/__init__.py:240: RemovedInDjango50Warning: The USE_L10N setting is deprecated. Starting with Django 5.0, localized formatting of data will always be enabled. For example Django will display numbers and dates using the format of the current locale.
    warnings.warn(USE_L10N_DEPRECATED_MSG, RemovedInDjango50Warning)
```

File: tests/testapp/settings.py [hunk 2]
```diff
@@ -37,5 +39,7 @@
...
 USE_I18N = True
+USE_TZ = True
```
Comment (adamchainz) [inline tests/testapp/settings.py:45]: fix deprecation warning about [default changing](https://docs.djangoproject.com/en/4.0/releases/4.0/#time-zone-support):

```
/.../django-select2/venv/lib/python3.10/site-packages/django/conf/__init__.py:214: RemovedInDjango50Warning: The default value of USE_TZ will change from False to True in Django 5.0. Set USE_TZ to False in your project settings if you want to keep the current default behavior.
    warnings.warn(
```

**Comment by adamchainz** [comment]
I managed to run the tests locally with the latest Chrome + chromedriver, pending approval to run on GitHub Actions.

**Comment by codingjoe** [comment]
Thanks, @adamchainz, appreciate the help 👍 
I'll invest some time to update some other things on the CI suite today. I'll ship a new release before the weekend 😃

**Pull request merged**